src/main/java/com/liferay/faces/util/cache/internal/CacheImpl.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2000-2017 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.util.cache.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.liferay.faces.util.cache.Cache;
+
+
+/**
+ * @author  Kyle Stiemann
+ */
+public class CacheImpl<K, V> implements Cache<K, V> {
```

Comment:
@stiemannkj1 If we decide that it is OK to say in the factory JavaDoc that the return value of `getCache()` "guaranteed to be Serializable then this class must `implement Serializable`